src/main/webapp/app/delete-dialog/delete-dialog-service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { DeleteDialogComponent } from 'app/delete-dialog/delete-dialog.component';
+
+/**
+ * Data that will be passed to the delete dialog component
+ */
+export class DeleteDialogData {
+    entityTitle: string;
+    deleteQuestion: string;
+    deleteConfirmationText?: string;
+}
+@Injectable({ providedIn: 'root' })
+export class DeleteDialogService {
+    constructor(private modalService: NgbModal) {}
+
+    /**
+     * Opens delete dialog and returns a result after dialog is closed
+     * @param deleteDialogData data that is used in dialog
+     */
+    openDeleteDialog(deleteDialogData: DeleteDialogData): Promise<any> {
+        const modalRef = this.modalService.open(DeleteDialogComponent, { size: 'lg', backdrop: 'static' });
+        modalRef.componentInstance.entityTitle = deleteDialogData.entityTitle;
+        modalRef.componentInstance.deleteQuestion = deleteDialogData.deleteQuestion;
+        modalRef.componentInstance.deleteConfirmationText = deleteDialogData.deleteConfirmationText;
+        return modalRef.result;
```

Comment:
I would prefer if we would expose an Observable here. This makes it easier to integrate the modal service into RxJS control flow.
https://www.learnrxjs.io/operators/creation/from.html

Reply:
I did it as an Observable, but for me Promise makes more sense here, as it only handles single event

Reply:
It is also a valid use case to use Observables for single emit streams, see e.g. the HttpClient from Angular. The main advantage is that you can integrate it more easily with RxJS control flow if it's an Observable.